ConfigSpace/hyperparameters/categorical.pyx

```diff
@@ -236,7 +236,7 @@ cdef class CategoricalHyperparameter(Hyperparameter):
     def check_default(self, default_value: Union[None, str, float, int]
                       ) -> Union[str, float, int]:
         if default_value is None:
-            return self.choices[0]
+            return self.choices[np.argmax(self.weights) if hasattr(self, "weights") else 0]
```

Comment:
Why the `hasattr` I'm on my phone so I can't check the code but I would have expected the attribute to always be there?

Reply:
For now, attribute `self.weights` exists only if `weights` is specified in the initialization. I'll modify that to make this attribute always defined, also so that it can be reused in `__copy__`